pyccel/ast/datatypes.py

```diff
@@ -603,7 +654,7 @@ def switch_basic_type(self, new_type):
         """
         assert isinstance(new_type, FixedSizeType)
```

Comment:
Given that this function only accepts `FixedSizeType` instances, the docstring should be updated.